## Especificación del Modelo. 
El modelo aumentado de Solow de Mankiw-Romer-Weil (MRW) se utiliza como base para este trabajo. La función de producción que incorpora el tamaño y la calidad del gobierno es de la forma Cobb-Douglas: 

(13) $y(t)=Ak(t)^α  h(t)^β  (g(t) e^(µθ) )^γ $

donde y(t) es la producción por trabajador; k(t) es el stock de capital privado por trabajador; y h(t) es el stock de capital humano por trabajador. La dimensión de tamaño del gobierno g(t), se mide por el stock de capital gubernamental por trabajador, y θ mide la dimensión de calidad del gobierno. Se asume la forma exponencial para la variable de calidad, ya que la buena gobernanza no es un factor de entrada directo, sino que sirve para mejorar la eficiencia del stock de capital del gobierno.

En el equilibrio de estado estacionario, hay una tasa exógena de progreso tecnológico ϖ y una tasa de crecimiento de la fuerza laboral n. El stock de capital se deprecia a una tasa δ. Si la fracción del ingreso dedicada al capital privado, el capital humano y el capital público son respectivamente, SK, SH y SG, el nivel de estado estable de la producción per cápita se alcanza cuando la adición al stock de cada tipo de capital es solo suficiente para satisfacer la tasa de crecimiento de la fuerza laboral y reemplazar el capital que se deprecia a una tasa de δ. El nivel de estado estacionario de producción per cápita en forma lineal logarítmica se puede expresar:
(14) $ln⁡[Y(t)/L(t) ]=a_0+a_(1 ) ln[S_K/(n+ϖ+ δ)]+a_2  ln[S_H/(n+ϖ+ δ)]+a_3  ln[S_G/(n+ϖ+ δ)]+a_4  θ $
Donde a_4=µγ.

La tasa de crecimiento del producto por trabajador en la transición al estado estacionario se puede expresar: 
(15) $ln⁡y(t)-ln⁡ y(0)=(1-e^(-λt))[ln⁡(y^* )-ln⁡y(0) ]$
donde y(0) es el nivel inicial de producción por trabajador e y* es el nivel de ingresos en estado estacionario por trabajador. λ es la velocidad de convergencia y además λ=(1-α-β-γ)(n+ϖ+ δ). Restando y(0) de ambos lados de la ecuación y sustituyendo por y* se obtiene el modelo de transición que se puede estimar como:

(16)  $ln⁡ y(t)-ln y(0)= a_0+a_(1 ) ln[S_K/(n+ϖ+ δ)]+a_2  ln[S_H/(n+ϖ+ δ)]+a_3  ln[S_G/(n+ϖ+ δ)]+a_4  θ+a_(5 )  ln⁡ y(0)+ ε  $

De acuerdo con la ecuación (16), la tasa de crecimiento del ingreso per cápita depende de la acumulación de capital privado, capital humano, capital público y de un buen gobierno. Aplicando el mismo razonamiento que Hulten (1996), un país con mejor gobierno convergerá a un mayor nivel de ingreso per capita estable que un país con mala calidad gubernamental. Si comienzan con el mismo nivel de ingreso per cápita, el país con una mejor gobernanza experimentará una tasa de crecimiento más rápida. 

## Metodología

El modelo que se utilizará para conocer el efecto de la calidad del gobierno y el gasto de este en el crecimiento es el modelo aumentado de Solow de Mankiw-Romer-Weil (MRW) visto en la sección anterior, pero a diferencia del ensayo original, los resultados se calcularan mediante una regresión de panel. 
La siguiente es la especificación de la ecuación: 

(15)  $ln⁡ (Y/N)=a_0+a_(1 )  ln⁡(S_K/(ngd))+a_2  ln⁡(S_H/ngd)+a_3  ln S_G/ngd+a_4  magobernanza+ a_5  agobernanza+ a_6  bgobernanza $

La descripción de las variables es		
* Y/N: Se utilizó la producción per cápita en lugar de la producción por trabajador al probar el modelo Solow. Mientras se pruebe el paradigma de Solow, se puede emplear la variable per cápita o por trabajador. MRW utilizó la producción por trabajador, mientras que Islam (1995) y Caselli (1996) utilizaron la variable per cápita.
* Sk: Porcentaje de la formación bruta de capital en el PBI real con base en la paridad de poder adquisitivo actual.
* Sh: Índice de capital humano por persona, basado en años de escolaridad y retornos a la educación. Es un proxy para la inversión en capital humano.
* Sg: Porcentaje del consumo del gobierno en el PBI real con base en la paridad de poder adquisitivo actual.
* ngd: es la suma de la tasa media de depreciación del stock de capital (d), la tasa de crecimiento de la tecnología (g) y la tasa de crecimiento de la población (n).  En este trabajo se supone que la suma de la tasa de crecimiento de la tecnología, g, y la tasa de depreciación, s, es 0.05 como en MRW. Si bien esta suposición puede no parecer apropiada para todos los países de la muestra, proporciona un punto de referencia útil para probar el modelo en cuestión. Este supuesto se ha utilizado en el trabajo de Aschauer (2000), Hulten (1996), Nazrul (1995) entre otros, tanto para las economías desarrolladas como en desarrollo.
* Variables Dummy de Calidad de Gobierno: se basan en la base de datos de gobierno compilada por Kaufmann, Kraay y Mastruzzi (2006) y que forma parte del World Bank's World Development Indicators. Kaufmann y col. han construido seis indicadores de gobernanza: (1) voz y responsabilidad: el grado en que los ciudadanos de un país pueden participar en el proceso de toma de decisiones políticas (2) estabilidad política y ausencia de violencia: mide la estabilidad de un gobierno ante la violencia política y el terrorismo (3) la efectividad del gobierno mide la capacidad de un gobierno para implementar políticas efectivas y mantener credibilidad (4) la calidad regulatoria es la capacidad del gobierno para formular e implementar políticas sólidas que fomenten la participación del sector privado (5) el imperio de la ley es la existencia de un buen sistema legal que incluye los derechos de propiedad y la ejecución de los contratos (6) el control de la corrupción mide el grado en que el poder público se desvía del beneficio privado. Las variables dummy de calidad de gobierno en el presente estudio se construyen de la siguiente manera. Los indicadores individuales (1, 2, 3, 4, 5 y 6) para cada año se promedian para construir un índice de gobernanza global compuesto (índice de calidad). Este índice de gobernanza compuesto varía de un valor de -2.5 a +2.5. Para distinguir entre los diferentes niveles de gobierno de muy alto a muy bajo, se identifican cuatro variables dicotómicas de gobierno:
	Θ ≥ 1.5 muy alta calidad de gobierno(magobernanza)
	1.5 > Θ > 0 alta calidad de gobierno (agobernanza)
	0 > Θ > -1.5 baja calidad de gobierno (bgobernanza)
	 Θ ≤ -1.5 muy baja calidad de gobierno (mbgobernanza)
	Donde Θ es el índice de calidad de gobierno compuesto.

Como la regresión de MCO no considera los efectos específicos del país no observados en el modelo (ec. 15), por lo tanto, la heterogeneidad del país puede aparecer en los parámetros estimados. En consecuencia, se calculará la ecuación 15 con efectos específicos de los países no observados usando last squeare dummy variable (LSDV) para comparar con los resultados con la regresión pooled. El uso de efectos no observados específicos de los países tiene una gran ventaja porque reduce los errores de medición.


In [1]:
# Importamos las librerías
import pandas as pd
import numpy as np
import math as mat
import statsmodels as sm
import statsmodels.api as sma
import statsmodels.formula.api as smf
import seaborn as sns
import scipy.stats as scs
from linearmodels import PanelOLS, FirstDifferenceOLS
from linearmodels.panel import compare

# Creamos un DataFrame en Pandas con los datos de la PWT que abarca desde 2002 hasta 2017
PWT= pd.read_csv("pwt91final.csv", sep=';')
PWT.columns

Index(['countrycode', 'country', 'currency_unit', 'year', 'pop', 'emp', 'hc',
       'ctfp', 'rgdpna', 'rtfpna', 'rwtfpna', 'delta', 'csh_c', 'csh_i',
       'csh_g', 'csh_x', 'csh_m', 'csh_r', 'n', 'l', 'Y/N', 'Y/L', 'var_yn',
       'var_yl', 'g', 'gobindex', 'Unnamed: 26', 'Unnamed: 27'],
      dtype='object')

In [2]:
# Indexamos el Dataframe segun paises y años 
PWT = PWT.set_index(['country','year'])
PWT

countrycode currency_unit        pop       emp        hc  \
country  year                                                            
Angola   2002         AGO        Kwanza  17.572649  7.729256  1.320135   
         2003         AGO        Kwanza  18.203369  8.002745  1.331887   
         2004         AGO        Kwanza  18.865717  8.254398  1.343743   
         2005         AGO        Kwanza  19.552542  8.791112  1.355706   
         2006         AGO        Kwanza  20.262400  9.355515  1.367774   
...                   ...           ...        ...       ...       ...   
Zimbabwe 2013         ZWE     US Dollar  15.054506  7.914061  2.504635   
         2014         ZWE     US Dollar  15.411675  8.222112  2.550258   
         2015         ZWE     US Dollar  15.777451  8.530669  2.584653   
         2016         ZWE     US Dollar  16.150362  8.839398  2.616257   
         2017         ZWE     US Dollar  16.529903  9.181251  2.648248   

                   ctfp         rgdpna    rtfpna   rwtfpna     delta  ...  \
country  year                                                         ...   
Angola   2002  0.191173   86755.960938  0.712762  0.730875  0.066037  ...   
         2003  0.185004   89349.828125  0.661461  0.718534  0.065295  ...   
         2004  0.200998   99136.398438  0.647934  0.684178  0.063845  ...   
         2005  0.251852  114035.796875  0.709638  0.704696  0.061244  ...   
         2006  0.334380  127318.640625  0.781533  0.739396  0.058514  ...   
...                 ...            ...       ...       ...       ...  ...   
Zimbabwe 2013  0.218595   29393.619141  1.150305  1.079854  0.036627  ...   
         2014  0.218886   30018.845703  1.137254  1.052321  0.035726  ...   
         2015  0.212050   30526.042969  1.118448  1.104721  0.034893  ...   
         2016  0.210578   30713.994141  1.095557  1.002600  0.034051  ...   
         2017  0.212403   31637.406250  1.091832  0.986483  0.033417  ...   

                  csh_m     csh_r         n         l          Y/N  \
country  year                                                        
Angola   2002 -0.143975 -0.197427  0.034704  0.035166  4936.988210   
         2003 -0.183415 -0.185301  0.035892  0.035384  4908.422580   
         2004 -0.214477 -0.091360  0.036386  0.031446  5254.843947   
         2005 -0.183027 -0.117501  0.036406  0.065022  5832.274823   
         2006 -0.183940 -0.050301  0.036305  0.064202  6283.492709   
...                 ...       ...       ...       ...          ...   
Zimbabwe 2013 -0.426188  0.090225  0.023362  0.039135  1952.479779   
         2014 -0.340442  0.051500  0.023725  0.038924  1947.798978   
         2015 -0.354298 -0.023353  0.023734  0.037528  1934.789328   
         2016 -0.310446  0.003050  0.023636  0.036190  1901.752674   
         2017 -0.299539  0.019133  0.023500  0.038674  1913.949856   

                        Y/L    var_yn    var_yl         g  gobindex  
country  year                                                        
Angola   2002  11224.361642  0.106887  0.106393  0.619834 -1.386467  
         2003  11164.898014 -0.005786 -0.005298 -0.071976 -1.248317  
         2004  12010.130149  0.070577  0.075704 -0.020450 -1.287372  
         2005  12971.714793  0.109885  0.080064  0.095232 -1.206835  
         2006  13608.940509  0.077366  0.049124  0.101311 -1.136141  
...                     ...       ...       ...       ...       ...  
Zimbabwe 2013   3714.100622  0.031206  0.015554  0.025016 -1.359576  
         2014   3650.989769 -0.002397 -0.016992 -0.011346 -1.317800  
         2015   3578.387839 -0.006679 -0.019886 -0.016537 -1.202702  
         2016   3474.670199 -0.017075 -0.028984 -0.020467 -1.216650  
         2017   3445.871126  0.006414 -0.008288 -0.003399 -1.229962  

[1840 rows x 24 columns]

In [3]:
# Renombramos y creamos variables necesarias para el modelo
PWT['sk'] = PWT['csh_i']
PWT['sh'] = PWT['hc']
PWT['sg'] = PWT['csh_g']

PWT['ngd'] = PWT['n'] + 0.05

PWT['sk/ngd']= PWT['sk']/PWT['ngd']
PWT['sh/ngd']= PWT['sh']/PWT['ngd']
PWT['sg/ngd']= PWT['sg']/PWT['ngd']

# Armamos las variables Dummy de calidad de Gobierno 
PWT['mbgovernanza'] = 0 
PWT.loc[(PWT.gobindex<-1.5), 'mbgovernanza'] = 1

PWT['bgovernanza'] = 0
PWT.loc[(PWT.gobindex>=-1.5) & (PWT.gobindex<0), 'bgovernanza']= 1

PWT['agovernanza'] = 0
PWT.loc[(PWT.gobindex>=0) & (PWT.gobindex<=1.5), 'agovernanza']= 1

PWT['magovernanza'] = 0
PWT.loc[ (PWT.gobindex>1.5), 'magovernanza']= 1

In [4]:
# Realizamos la regresion Pooled
pooled = PanelOLS.from_formula("np.log(PWT['Y/N'])~ 1 + np.log(PWT['sk/ngd']) + np.log(PWT['sh/ngd']) + np.log(PWT['sg/ngd']) + PWT['magovernanza'] + PWT['agovernanza'] + PWT['bgovernanza'] ", data=PWT).fit(cov_type="robust", use_lsdv=True)
print(pooled)

                          PanelOLS Estimation Summary                           
Dep. Variable:     np.log(PWT['Y/N'])   R-squared:                        0.6799
Estimator:                   PanelOLS   R-squared (Between):              0.7174
No. Observations:                1840   R-squared (Within):              -1.7816
Date:                Mon, Mar 09 2020   R-squared (Overall):              0.6799
Time:                        09:49:57   Log-likelihood                   -1847.2
Cov. Estimator:                Robust                                           
                                        F-statistic:                      648.92
Entities:                         115   P-value                           0.0000
Avg Obs:                       16.000   Distribution:                  F(6,1833)
Min Obs:                       16.000                                           
Max Obs:                       16.000   F-statistic (robust):             925.04
                            

El panel de la tabla 1 muestra los resultados del modelo. En la regresión de panel pooled todas las variables son significativas con un nivel de significancia de 0,01. Las variables dummy de calidad de gobierno donde el grupo de referencia es el de muy baja calidad, muestran que mientras mayor sea la gobernanza de los países mayor será el ingreso per capita de estos. Y los coeficientes estimados sugieren que los países con “muy alta” calidad de gobierno posen un [100(e1,76-1)] ≈ 481,17% más de ingresos per cápita que los países con “muy mala” gobernanza, aquellos con “alta” calidad de gobierno disponen un 297,49% más que estos y por ultimo las economías con “baja” gobernanza cuentan con un 68,20% mayor de producción per capita que el grupo de peor calidad de gobierno. La elasticidad del ingreso per capita respecto a el consumo del gobierno es negativa y significaría que un aumento de este en un 1% disminuye PBI real per cápita en 0,81%. Los coeficientes estimados del capital privado y el capital humano son positivos e indican que un incremento del 1% del primero aumenta el ingreso per cápita 0,52% mientras que el segundo adiciona un 1,5%. 

In [5]:
# Realizamos la regresion con efectos fijos LSDV
lsdv = PanelOLS.from_formula("np.log(PWT['Y/N'])~ 1 + EntityEffects  + np.log(PWT['sk/ngd']) + np.log(PWT['sh/ngd']) + np.log(PWT['sg/ngd']) + PWT['magovernanza'] + PWT['agovernanza'] + PWT['bgovernanza'] ", data=PWT).fit(cov_type="robust",use_lsdv=True)
print(lsdv)

                          PanelOLS Estimation Summary                           
Dep. Variable:     np.log(PWT['Y/N'])   R-squared:                        0.1543
Estimator:                   PanelOLS   R-squared (Between):              0.2706
No. Observations:                1840   R-squared (Within):               0.1543
Date:                Mon, Mar 09 2020   R-squared (Overall):              0.2688
Time:                        09:49:57   Log-likelihood                    1122.3
Cov. Estimator:                Robust                                           
                                        F-statistic:                      52.272
Entities:                         115   P-value                           0.0000
Avg Obs:                       16.000   Distribution:                  F(6,1719)
Min Obs:                       16.000                                           
Max Obs:                       16.000   F-statistic (robust):             23.328
                            

La tabla 2 muestra los resultados usando los estimadores LSDV para incorporar los efectos no observables de los países en la regresión. Todas las variables poseen significancia para un nivel de 0,01. Al igual que en la regresión pooled las variables dummy muestran que cuanto mayor sea la calidad de gobierno mayor será el ingreso per cápita de los países ya que sus coeficientes sugieren que los países con gobernanza 'muy alta' sus ingreso per cápita son [100 (e0.3954-1)] ≈ 48,50% mayores que los países con gobernanza 'muy baja', y aquellos con gobernanza 'alta' poseen  45.60% más de producción per cápita , y los países con 'baja' ' calidad de gobierno disponen  39,78% más que los países con una gobernanza 'muy baja '. La elasticidad del ingreso per capita respecto a el consumo del gobierno es negativa y significaría que un aumento de este en un 1% disminuye PBI real per cápita en 0,07%. Los coeficientes estimados del capital privado y el capital humano son positivos e indican que un incremento del 1% del primero aumenta el ingreso per cápita 0,12% mientras que el segundo adiciona un 0,48%


In [6]:
# Tabla Comparativa
print(compare({'1.Pooled':pooled,'2.LSDV':lsdv}))

                            Model Comparison                           
                                        1.Pooled                 2.LSDV
-----------------------------------------------------------------------
Dep. Variable                 np.log(PWT['Y/N'])     np.log(PWT['Y/N'])
Estimator                               PanelOLS               PanelOLS
No. Observations                            1840                   1840
Cov. Est.                                 Robust                 Robust
R-squared                                 0.6799                 0.1543
R-Squared (Within)                       -1.7816                 0.1543
R-Squared (Between)                       0.7174                 0.2706
R-Squared (Overall)                       0.6799                 0.2688
F-statistic                               648.92                 52.272
P-value (F-stat)                          0.0000                 0.0000
=======================     ====================   =============

## Conclusión 

Este estudio examina los efectos del tamaño y la calidad del gobierno sobre el desarrollo económico en 115 países para el periodo 2002 al 2017. Los resultados indican que el aumento del gasto público disminuye el ingreso per cápita, aunque de forma muy pequeña pero el mejoramiento de la calidad de gobierno puede mejorar los resultados de este. El capital humano como el físico también afectan significativa y positivamente el desarrollo económico. Los resultados son consistentes con los de Hulton, Aschauer y Prichett, quienes muestran que mejorar la eficacia del capital público mediante una mejora de la calidad de gobierno puede conducir a un mejor crecimiento. En conclusión, se puede afirmar que los países deben promover la buena gobernanza para crecer y desarrollarse ya que es necesaria para poseer un gobierno eficiente que evite el despilfarro de fondos públicos que perjudican al desarrollo económico.


## Apéndice 

Lista de 115 países utilizados en el trabajo : Angola, Alemania, Nicaragua, Tanzania, Argentina, Grecia, Níger, Ucrania, Armenia, Guatemala, Nigeria, Reino Unido, Australia, Honduras, Noruega, Estados Unidos, Austria, Hungría, Panamá, Uruguay, Baréin, Islandia, Paraguay, Venezuela, Barbados, India, Perú, Zimbabue, Bélgica, Indonesia, Filipinas, Benín, Irán, Polonia, Bolivia, Irak, Portugal, Botsuana, Irlanda, Catar, Brasil, Israel, República de Corea, Bulgaria, Italia, Moldavia, Burkina Faso, Jamaica, Rumania Burundi, Japón, Rusia,	 Camerún, Jordania, Ruanda, Canadá, Kazajistán, Arabia Saudita, República Centroafricana, Kenia, Senegal, Chile, Kuwait, Serbia, China, Kirguistán, Sierra Leona, Hong Kong, Laos, Singapur, Macao, Letonia, Eslovaquia, Colombia, Lesoto, Eslovenia, Costa Rica, Lituania, Sudáfrica, Costa de Marfil, Luxemburgo, España, Croacia, Malasia, Sri Lanka, Chipre, Malta, Sudán, Dinamarca, Mauritania, Suecia, República Dominicana, Mauricio, Suiza, Ecuador, México, Taiwán, Egipto, Mongolia, Tayikistán, Estonia, Marruecos, Tailandia, Fiyi, Mozambique, Togo, Finlandia, Namibia, Trinidad y Tobago, Francia, Países Bajos, Tunicia, Gabón, Nueva Zelanda, Turquía. 


## Bibliografía 

Aschauer, D. A. (2000). Public capital and economic growth: issues of quantity, finance, and efficiency. Economic Development and Cultural Change, 48(2), 391-406.

Barro, R. J. (1990). Government spending in a simple model of endogeneous growth. Journal of political economy, 98(5, Part 2), S103-S125.

Barro, R. J. (1991). Economic growth in a cross section of countries. The quarterly journal of economics, 106(2), 407-443.

Barth, J. R., & Bradley, M. D. (1988). The impact of government spending on economic activity. National Chamber Foundation.

Das, D. K. (2013). Empirical Estimation of the Solow Growth Model: A Panel Approach.
Devarajan, S, Swaroop, V and Zou, H. (1996). “The Composition of Public Expenditure and Economic Growth.” Journal of Monetary Economics, 37, 313-344.

Easterly, W., & Rebelo, S. (1993). Fiscal policy and economic growth. Journal of monetary economics, 32(3), 417-458.

Grier, K. B., & Tullock, G. (1989). An empirical analysis of cross-national economic growth, 1951–1980. Journal of monetary economics, 24(2), 259-276.

Hulten, C. R. (1996). Infrastructure capital and economic growth: How well you use it may be more important than how much you have (No. w5847). National Bureau of Economic Research.

Islam, N. (1995). Growth empirics: a panel data approach. The quarterly journal of economics, 110(4), 1127-1170.
Kormendi, R. C., & Meguire, P. G. (1985). Macroeconomic determinants of growth: cross-country evidence. Journal of Monetary economics, 16(2), 141-163.

Landau, D. (1983). Government expenditure and economic growth: a cross-country study. Southern Economic Journal, 783-792.

Mankiw, N. G., Romer, D., & Weil, D. N. (1992). A contribution to the empirics of economic growth. The quarterly journal of economics, 107(2), 407-437.

Pritchett, L. (1999). Mind your p's and q's: the cost of public investment is not the value of public capital. The World Bank.

Solow, R. M. (1956). A contribution to the theory of economic growth. The quarterly journal of economics, 70(1), 65-94.
